lib/services/role.go

```diff
@@ -2589,18 +2595,18 @@ func (set RoleSet) checkAccess(r AccessCheckable, traits wrappers.Traits, state
 		if !matchNamespace {
 			continue
 		}
-
-		matchLabels, labelsMessage, err := checkRoleLabelsMatch(types.Deny, role, traits, r, isDebugEnabled)
-		if err != nil {
-			return trace.Wrap(err)
-		}
-		if matchLabels {
-			debugf("Access to %v %q denied, deny rule in role %q matched; match(namespace=%v, %s)",
-				r.GetKind(), r.GetName(), role.GetName(), namespaceMessage, labelsMessage)
-			return trace.AccessDenied("access to %v denied. User does not have permissions. %v",
-				r.GetKind(), additionalDeniedMessage)
+		if requiresLabelMatching {
```

Comment:
Here and below, let's add an explicit log entry that would say the label matching is being skipped for resource X or something along those lines.